Tools (#488)
elevenlabs/elevenlabs-docs: product/conversational-ai/tools.mdx

---
title: "Tools"
description: "Provide your agent with real time information and the ability to take action in third party apps with external function calls."
---

Tools allow you to make external function calls to third party apps so you can get real-time information. You might use tools to:
- Schedule appointments on someone's calendar
- Book restaurant reservations

> - Create or update customer records in a Customer

Reviewer: ```suggestion
- Create or update customer records in a CRM system
```

You can find Secrets within the Conversational AI Dashboard in the Agent subnav. 

<img height="200" src="/conversational-ai/images/bearer_secret.png" />

### Webhooks
Next, look for "Tools" in the "Agent" subnav. Add a new Tool to configure your webhook. For our AI receptionist, we created two Tools to interact with the Cal.com API:

(1) We want our AI receptionist to be able to answer questions like "When is Sam available to meet on tomorrow?" or "Is Sam free at 10:30am on Tuesday?". To make this possible, we integrated the Cal.com [Get Available Slots](https://cal.com/docs/api-reference/v2/slots/get-available-slots#get-available-slots) endpoint.

Name: Get_Available_Slots

Description: This API takes a date / time range as input and returns open calendar slots within that range.

Method: Get

URL: https://api.cal.com/v2/slots/available

(2) Once our caller and receptionist have agreed on a suitable time to meet, we'd like to book a meeting. For this we use the [Create a booking](https://cal.com/docs/api-reference/v2/bookings/create-a-booking#create-a-booking) endpoint. 

Name: Book_Meeting

> Description: Call this function once you have collect the caller's full name, meeting time, and email address.

Reviewer: ```suggestion
Description: Call this function once you have collected the caller's full name, meeting time, and email address.
```

<img height="200" src="/conversational-ai/images/cal_book_meeting.png" />

In our Description for each, we define a prompt that our Conversational Agent will use to extract the relevant information from the call transcript using an LLM.

Here's how we defined our body parameters for our AI receptionist:
| Identifier  | Data Type | Required | Description                                                                                                                                                                                      |
|-------------|-----------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| startTime   | String    | Yes      | The start time of the slot the person is checking availability for in UTC timezone, formatted as ISO 8601 (e.g., '2024-08-13T09:00:00Z'). Extract time from natural language and convert to UTC. |

> | eventTypeId | String    | Yes      | The desired meeting length, as an event id. If 15 minutes, return 1351800. If 30 minutes, return 1351801. If 60 minutes, return 1351802. If you are not sure, return 1351801.                    |

Reviewer: ```suggestion
| eventTypeId | String    | Yes      | The unique Cal event ID for the meeting duration. Use 1351800 for a 15-minute meeting, 1351801 for 30 minutes, and 1351802 for 60 minutes. If no specific duration is provided, default to 1351801 (30 minutes).                   |
```
